contracts/token/ERC721/ERC721BasicToken.sol

```diff
@@ -0,0 +1,279 @@
+pragma solidity ^0.4.18;
+
+import "./ERC721Basic.sol";
+import "./ERC721Receiver.sol";
+import "../../math/SafeMath.sol";
+import "../../AddressUtils.sol";
+
+/**
+ * @title ERC721 Non-Fungible Token Standard basic implementation
+ * @dev see https://github.com/ethereum/eips/issues/721
+ */
+contract ERC721BasicToken is ERC721Basic {
+  using SafeMath for uint256;
+  using AddressUtils for address;
+  
+  // Equals to bytes4(keccak256("onERC721Received(address,uint256,bytes)"))
+  bytes4 ERC721_RECEIVED = 0xf0b9e5ba; 
+
+  // Mapping from token ID to owner
+  mapping (uint256 => address) internal tokenOwner;
+
+  // Mapping from token ID to approved address
+  mapping (uint256 => address) internal tokenApprovals;
+
+  // Mapping from owner to number of owned token
+  mapping (address => uint256) internal ownedTokensCount;
+
+  // Mapping from owner to operator approvals
+  mapping (address => mapping (address => bool)) internal operatorApprovals;
+
+  /**
+  * @dev Guarantees msg.sender is owner of the given token
+  * @param _tokenId uint256 ID of the token to validate its ownership belongs to msg.sender
+  */
+  modifier onlyOwnerOf(uint256 _tokenId) {
+    require(ownerOf(_tokenId) == msg.sender);
+    _;
+  }
+
+  /**
+  * @dev Checks msg.sender can transfer a token, by being owner, approved, or operator
+  * @param _tokenId uint256 ID of the token to validate
+  */
+  modifier canTransfer(uint256 _tokenId) {
```

Comment:
should this be `isApprovedFor(_tokenId)` because it can be used for things beyond transfer?

Reply:
I kind of like the name because it clearly distinguishes from `isApproved` (which van be interpreted as "has been individually approved) and `isApprovedForAll` or `isOwner`

Reply:
Actually it's only being used to checking transfers. Where else do you see this being used @Shrugs? We could change it to `canManage` to appease all use cases.

Reply:
perhaps a superclass will want to allow an owner to change the metadata of the token on-chain? (or is that not allowed by the spec?). Regardless, you're right; the concept of ownership here is pretty much just "can I transfer this token" which seems fine. I'm not particularly attached to `canManage` but I feel like it's slightly more general without losing meaning.